toolkit/tools/pkg/imagecustomizerlib/customizeoverlayfs.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/file"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+)
+
+func enableOverlayFS(overlays *[]imagecustomizerapi.OverlayFS, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	if overlays == nil {
+		return nil
+	}
+
+	// Integrate overlayfs dracut module and overlay driver into initramfs img.
+	// Integrate systemd veritysetup dracut module into initramfs img.
+	overlayfsDracutModule := "overlayfs"
+	overlayDracutDriver := "overlay"
+	err = buildDracutModule(overlayfsDracutModule, overlayDracutDriver, imageChroot)
+	if err != nil {
+		return err
+	}
+
+	// Dereference the pointer to get the slice
+	overlayFSConfigs := *overlays
+	err = updateGrubConfigForOverlayFS(imageChroot, overlayFSConfigs)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func updateGrubConfigForOverlayFS(imageChroot *safechroot.Chroot, overlays []imagecustomizerapi.OverlayFS) error {
+	var err error
+	var newArgsParts []string
+
+	// Iterate over each OverlayFS configuration
+	for _, overlay := range overlays {
+		formattedPersistentPartition, err := systemdFormatPartitionId(overlay.PersistentPartition.IdType, overlay.PersistentPartition.Id)
+		if err != nil {
+			return err
+		}
+
+		// Construct cmdline arguments for each OverlayFS
+		newArg := fmt.Sprintf(
+			"rd.overlays=%s,%s,%s rd.overlayfs_persistent_volume=%s",
```

Comment:
I am thinking to make `rd.overlays` accept four values, like `rd.overlays=%s,%s,%s,%s`, the fourth %s will take partition. Idea?

Reply:
For now, users can provide more than one overlay configs, for example if they want multiple dirs to be overlay enabled, that is fine and our current overlay dracut module can achieve that. However, those multiple overlays will be either located at the same partition to be persistent, or they will be all volatile. That being said, the current overlay dracut module did not support multiple overlays locating at different partitions or supporting coexistence of volatile overlays & persistent overlays.

However, in the upcoming future, I plan to fix this and add these supports. So that is the reason why I want this format `rd.overlays=%s,%s,%s,%s` so that each partition will follow its overlaydirs.

Please advise and appreciate your ideas. Thanks.